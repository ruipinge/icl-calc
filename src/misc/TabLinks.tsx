import { NavLink } from 'react-router-dom';

interface Link {
  to: string;
  label: string;
}

const LINKS: Link[] = [
  {
    to: '/',
    label: 'Patient'
  },
  {
    to: '/normality',
    label: 'Biometric Normality'
  },
  {
    to: '/matrix',
    label: 'Floating Matrix'
  },
  {
    to: '/regression',
    label: 'Regression'
  }
];

export const TabLinks = () => (
  <ul className="nav nav-pills" style={{ marginBottom: '1rem' }}>
    {LINKS.map((link, index) => (
      <li className="nav-item" key={index}>
        <NavLink
          exact={true}
          className="nav-link"
          activeClassName="active"
          to={link.to}
        >
          {link.label}
        </NavLink>
      </li>
    ))}
  </ul>
);
