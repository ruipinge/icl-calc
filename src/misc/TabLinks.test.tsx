import { HashRouter as Router } from 'react-router-dom';
import { TabLinks } from './TabLinks';
import { render } from '@testing-library/react';

it('renders without crashing', () => {
  const { asFragment } = render(
    <Router>
      <TabLinks />
    </Router>
  );
  expect(asFragment()).toMatchSnapshot();
});
