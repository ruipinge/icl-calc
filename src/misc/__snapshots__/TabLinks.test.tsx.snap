// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <ul
    class="nav nav-pills"
    style="margin-bottom: 1rem;"
  >
    <li
      class="nav-item"
    >
      <a
        aria-current="page"
        class="nav-link active"
        href="#/"
      >
        Patient
      </a>
    </li>
    <li
      class="nav-item"
    >
      <a
        class="nav-link"
        href="#/normality"
      >
        Biometric Normality
      </a>
    </li>
    <li
      class="nav-item"
    >
      <a
        class="nav-link"
        href="#/matrix"
      >
        Floating Matrix
      </a>
    </li>
    <li
      class="nav-item"
    >
      <a
        class="nav-link"
        href="#/regression"
      >
        Regression
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
